### PR TITLE
Add timeout option to debug command

### PR DESCRIPTION
**Pull request by Fatme**
https://github.com/NativeScript/nativescript-cli/issues/682
https://github.com/NativeScript/nativescript-cli/issues/636
https://github.com/NativeScript/nativescript-cli/issues/375

Should be merged after this PR https://github.com/telerik/mobile-cli-lib/pull/406

### CODE REVIEW

File: lib/services/android-debug-service.ts [hunk 1]
```diff
@@ -195,8 +196,8 @@ class AndroidDebugService implements IDebugService {
     private startAndGetPort(packageName: string): IFuture<number> {
         return (() => {
             let port = -1;
-			let timeout = 60;
-
+			let timeout = this.$utils.getParsedTimeout(60);
```
Comment (rosen-vladimirov) [inline lib/services/android-debug-service.ts]: tab/space isssue maybe

File: lib/services/ios-debug-service.ts [hunk 2]
```diff
@@ -137,12 +145,14 @@ class IOSDebugService implements IDebugService {
...
-                    awaitNotification(npc, notification.appLaunching(projectId), 60000).wait();
+                    let timeout = this.$utils.getMilliSecondsTimeout(IOSDebugService.TIMEOUT_SECONDS);
+                    awaitNotification(npc, notification.appLaunching(projectId), timeout).wait();
                     process.nextTick(() => {
                         npc.postNotificationAndAttachForData(notification.waitForDebug(projectId));
                         npc.postNotificationAndAttachForData(notification.attachRequest(projectId));
                     });
-                    awaitNotification(npc, notification.readyForAttach(projectId), 5000).wait();
+                    
+                    awaitNotification(npc, notification.readyForAttach(projectId), this.getReadyForAttachTimeout(timeout)).wait();
```
Comment (rosen-vladimirov) [inline lib/services/ios-debug-service.ts]: `timeout / 10` maybe ?

File: lib/services/ios-debug-service.ts [hunk 3]
```diff
@@ -163,11 +173,12 @@ class IOSDebugService implements IDebugService {
                 let projectId = this.$projectData.projectId;
                 let npc = new iOSProxyServices.NotificationProxyClient(iosDevice, this.$injector);
                 
+                let timeout = this.getReadyForAttachTimeout();
                 let [alreadyConnected, readyForAttach, attachAvailable] = [
                     notification.alreadyConnected(projectId),
                     notification.readyForAttach(projectId),
                     notification.attachAvailable(projectId)
-                ].map((notification) => awaitNotification(npc, notification, 2000));
+                ].map((notification) => awaitNotification(npc, notification, timeout));
```
Comment (rosen-vladimirov) [inline lib/services/ios-debug-service.ts]: `timeout / 10` maybe ?

File: lib/services/ios-debug-service.ts [hunk 4]
```diff
@@ -183,7 +194,7 @@ class IOSDebugService implements IDebugService {
                         this.$errors.failWithoutHelp("A debugger is already connected.");
                     case attachAvailable:
                         process.nextTick(() => npc.postNotificationAndAttachForData(notification.attachRequest(projectId)));
-                        try { awaitNotification(npc, notification.readyForAttach(projectId), 2000).wait(); }
+                        try { awaitNotification(npc, notification.readyForAttach(projectId), timeout).wait(); }
```
Comment (rosen-vladimirov) [inline lib/services/ios-debug-service.ts]: `timeout / 10` maybe ?

**Comment by rosen-vladimirov** [comment]
:+1:

File: lib/services/ios-debug-service.ts [hunk 5]
```diff
@@ -244,6 +255,14 @@ class IOSDebugService implements IDebugService {
             return inspectorPath;
         }).future<string>()();
     }
+
+    
+    private getReadyForAttachTimeout(timeoutInMilliseconds?: number): number {
+        let timeout = timeoutInMilliseconds || this.$utils.getMilliSecondsTimeout(IOSDebugService.TIMEOUT_SECONDS);
+        let readyForAttachTimeout = timeout / 10 ;
+        let defaultReadyForAttachTimeout = 5000;        
+        return readyForAttachTimeout > defaultReadyForAttachTimeout ? readyForAttachTimeout : defaultReadyForAttachTimeout;
```
Comment (teobugslayer) [inline lib/services/ios-debug-service.ts]: You can use (Math.max)[https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/max] here

**Pull request merged**